### ISSUE TITLE
EUI QA

### ISSUE THREAD

**Issue by hrishikeshpaul** [hubmapconsortium/ccf-ui#547]
- [x]  Logo Bar has the correct height and logotype. Links to Data Portal. Login button is optional, not required. Should not be included unless necessary. The button does not have any of the hover states.
Hover:
background-color: #E0E0E0;
box-shadow: rgba(0, 0, 0, 0.2) 0px 5px 5px -3px, rgba(0, 0, 0, 0.14) 0px 8px 10px 1px, rgba(0, 0, 0, 0.12) 0px 3px 14px 2px;
- [ ]  Default state looks mostly good. However, Organ Carousel and Ontology don't match for what is active/inactive. They should be the same.
- [x]  Organ Carousel arrows should have the Not-allowed cursor when disabled. Disabled color should be #CCCCCC and active color #636363. When in ready state, cursor should be pointer.
- [x]  L Drawer: area at top looks good. The border color around the search should be #212121.
- [x]  Info icon should be the solid version, not the outline. It also doesn't have the hover state the other icons have and should.
- [x]  Reset icon next to theme icon is missing tooltip.
- [x]  Organ Carousel. Items are squished. While the surround container is fine at 80px, adjustments need to be made to the interior. Currently, a margin of .5rem is set on carousel-item. Change these to padding. This will make for a more square hover state with the gray background.
- [x]  Organ Carousel. mat-icon should be sized down to add some space for the numbers at bottom. Go from 2.75rem to 2rem. Also, currently padding is applied to the carousel container. Don't do that. Apply the padding to the individual organ tiles. That way the hover state extends top to bottom. You have multiple rails currently of nested containers creating space around the organ content (selector-drawer, then ccf-organ-selector, then container, then carousel-item-container). For some reason mat-icon is outside this last. That last should extend top to bottom of the drawer.
- [x]  R Drawer. The icon hover state is bad. Not a fan. The problem is how they are constructed. They are 24px x 24px. While technically correct for the icon itself, the hover state exists in addition to this space. The button that houses the icons is big enough to do what is needed. Increase the size of the mat-icon to 32px x 32px. Add 4px padding. Change border-radius to 16px to give us a round shape. AMENDMENT: The hover states used in RUI for icons are correct. Apply the same here. Refer to RUI for example.
- [ ]  Logo Bar: during page build when the spinner is taking a vacation on the screen, the Login button appears on the right, but the logotype sometimes doesn't show for days. When the login button is removed this means the user will see an empty bar. The logotype should be appear in the Logo Bar from the start. NOTE: This behavior is not consistent. Inlining SVG would solve the problem if a link is used. If you want to use a link, preload the logo.
- [ ]  Opacity: Cute push/pull animation. Works for the context. Advise adding a fade to the text though as it moves. The hover behavior for these is different, don't include a gray hover with the icon. The controls is the hover action. We don't need to states. AMENDMENT: The animation in RUI is superior. Use the same setup as there. Refer to RUI for an example.
- [x]  Drawer tabs. Use the #E0E0E0 color for the hover, not the blue. Advise adding fade to the Organ Carousel on open close animation. AMENDMENT: Drawer tabs are correctly built in RUI, simply reuse those.
- [ ]  L Drawer: Filter flyout looks good. The reset though bothers me. You have a global reset but multiple sections. I'd prefer the reset be associated with Assay Types and another with Tissue Providers. The blue for the checkboxes should be #636363 to match Data Portal. The Apply Filters button has no hover state but should (AMENDMENT: See RUI for an example of button hover state). There is a positioning issue for the select item container, it looks too far left. Age flyout shows some clipping on the right "0". There is opening animation for the Filters flyout but not closing. Closing should be a reverse of the opening animation.
- [ ]  Ontology: As stated, active/inactive items and their states should match with Organ Carousel.
- [x]  Stage: Rounded corners are in place at top but not at bottom. Borders sized appropriately. Wondering about a lower opacity for the blue bodies. Maybe drop another 10%. Try it, see which works better. AMENDMENT: RUI is correctly setup with rounded corners and correct framing of black area. Refer to RUI for an example of how to set up.
- [x]  R Drawer: The results area extends off the page. Seems to have a fixed height. The results should never exceed a bottom padding/margin of 24px. In effect, the results end where Stage does.
- [ ]  Stage: I'm seeing color coding for registration cubes. No legend. Why are they different colors? I thought we have only one color for blocks, which I thought was white.. Registration zones should have an even lower opacity, maybe 10%.
- [ ]  Interaction:
- [ ]  Disabling items in the Organ Carousel have no impact. I should see organs removed/added on Stage. The Ontology should update to show what is happening as well. E.g. If I select Kidney, R in the carousel then that should be active in the Ontology as well as on Stage.
- [ ]  Clicking in the Ontology seems to have no impact on the Stage.
- [ ]  Opactiy slider default state seems to Visibility Off. It should be Visibility On and Opacity 20%.
- [ ]  When Opacity aniamtion closes it resets the control to the default state of Visibility Off. This is not correct. It should retain either the default state of Visibility On and Opacity 20%, or the user defined input.
- [ ]  Ontology toggles. Would be nice to have some animation for the show/hide. Makes the transition less jarring. The arrow should animate in rotation. The text that is showing/hiding should at the least fade on and off.
- [ ]  Interacting with the models on Stage is painful. Requires the user knowing that zooming happens relative to where the cursor is pointed. I'm familiar because 3ds Max does the same thing. Certain zooming ends up with a black screen when it shouldn't. Out of a dozen attempts, only once did I get close enough to see the organ in detail.
- [ ]  When zoomed in an organ, my expectation is that I can see the organ. Move the organ in 3D space. That is not the case. It is the body that is moving.
- [ ]  Bruce, since the skin is now an organ, shouldn't that be in the Organ Carousel? That would empower the user to disable it entirely.

**Comment by hrishikeshpaul** [comment]
- [ ] Change loading text to 'Loading CCF Exploration User Interface (EUI)'
- [ ] Extend the loading screen to cover the whole page
